### PR TITLE
feat: Add default content for extension points to the outline in CPE

### PR DESCRIPTION
**Pull request by nikmace**
Feat for #1866.

- Enhances outline nodes transformation for extension points that have default content.
- Add default content to the outline tree in CPE.
- Shows information message to the user if extension point has default content

### CODE REVIEW

File: packages/control-property-editor/src/panels/outline/Tree.tsx
Comment (GDamyanov) [inline packages/control-property-editor/src/panels/outline/Tree.tsx:302]: Shouldn't we use texts from i18n?

File: packages/control-property-editor/src/panels/outline/Tree.tsx
Comment (GDamyanov) [inline packages/control-property-editor/src/panels/outline/Tree.tsx:394]: Shouldn't we use texts from i18n?

File: packages/preview-middleware-client/src/adp/controllers/ExtensionPoint.controller.ts
Comment (GDamyanov) [inline packages/preview-middleware-client/src/adp/controllers/ExtensionPoint.controller.ts:101]: We can return directly `defaultContent?.length > 0`

**Review by GDamyanov — CHANGES_REQUESTED** [review]

File: packages/preview-middleware-client/src/adp/controllers/ExtensionPoint.controller.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/adp/controllers/ExtensionPoint.controller.ts:101]: In this case, defaultContent can be undefined so I still need to check whether we have it and then check its length. But I changed it a little here: [c86f608](https://github.com/SAP/open-ux-tools/pull/1867/commits/c86f608325dd2d9f9b520beffc14d1a6122242c5)

File: packages/control-property-editor/src/panels/outline/Tree.tsx
Comment (nikmace) [reply packages/control-property-editor/src/panels/outline/Tree.tsx:398]: Added in: [c86f608](https://github.com/SAP/open-ux-tools/pull/1867/commits/c86f608325dd2d9f9b520beffc14d1a6122242c5)

File: packages/control-property-editor/src/panels/outline/Tree.tsx
Comment (nikmace) [reply packages/control-property-editor/src/panels/outline/Tree.tsx:304]: Added in: [c86f608](https://github.com/SAP/open-ux-tools/pull/1867/commits/c86f608325dd2d9f9b520beffc14d1a6122242c5)

**Review by GDamyanov — CHANGES_REQUESTED** [review]
Comments ad

File: packages/control-property-editor/src/panels/outline/Tree.tsx
Comment (nikmace) [reply packages/control-property-editor/src/panels/outline/Tree.tsx:303]: For some reason, onRenderCell handler has an item as optional. Well, it is because the nestingDepth is optional but we do not use it, then the item should also be optional because: `A required parameter cannot follow an optional parameter.`

```ts
const onRenderCell = (nestingDepth?: number, item?: OutlineNodeItem, itemIndex?: number): React.ReactNode => {
```

I do think that we will always have a name no matter if this item is optional. Just to be safe this check should stay?

**Review by GDamyanov — DISMISSED** [review]
Approve again after base branch merge.

**Review by voicis — CHANGES_REQUESTED** [review]

File: packages/preview-middleware-client/src/cpe/outline/nodes.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/cpe/outline/nodes.ts]: Changed in: [e856c19](https://github.com/SAP/open-ux-tools/pull/1867/commits/e856c19f239ee47c93afa3875e74b4ace810578c)

File: packages/preview-middleware-client/src/cpe/outline/nodes.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/cpe/outline/nodes.ts]: Changed in: [e856c19](https://github.com/SAP/open-ux-tools/pull/1867/commits/e856c19f239ee47c93afa3875e74b4ace810578c)

File: packages/preview-middleware-client/src/cpe/outline/index.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/cpe/outline/index.ts]: Changed in: [e856c19](https://github.com/SAP/open-ux-tools/pull/1867/commits/e856c19f239ee47c93afa3875e74b4ace810578c)

File: packages/preview-middleware-client/src/cpe/outline/index.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/cpe/outline/index.ts]: Changed in: [e856c19](https://github.com/SAP/open-ux-tools/pull/1867/commits/e856c19f239ee47c93afa3875e74b4ace810578c)

File: packages/preview-middleware-client/src/cpe/outline/index.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/cpe/outline/index.ts]: Added a comment for this method: [e856c19](https://github.com/SAP/open-ux-tools/pull/1867/commits/e856c19f239ee47c93afa3875e74b4ace810578c)

File: packages/preview-middleware-client/src/cpe/outline/nodes.ts
Comment (nikmace) [reply packages/preview-middleware-client/src/cpe/outline/nodes.ts]: Screenshot 2024-05-10 at 12 52 59

I'll explain it by referencing this screenshot. Extension points are usually under some control, in this case, it is Panel. We can see that Panel has two extension points. ExtensionPoint2 has no default content nor created controls ExtensionPoint3 has default content and we use `addChildToExtensionPoint` to add children to it. defaultContent is an array of control IDs, therefore we need to construct those children ourselves. 

When you add a fragment at ExtensionPoint3 (a button for example) it will no longer display the default content and it will be an empty array. On the other hand, created controls will have an array with IDs, in our case, the id of the button.
Screenshot 2024-05-10 at 13 28 37

Both default content and created controls are controls and have a parent so if we do not remove them they will be rendered on the same level as the two extension points and one Label under Panel. We want to visualize them under the extension point to which they belong. That is why we need to remove those duplicate nodes and that is why we keep a unique set of IDs to later remove them.

File: packages/preview-middleware-client/src/cpe/outline/nodes.ts
Comment (voicis) [reply packages/preview-middleware-client/src/cpe/outline/nodes.ts]: @nikmace Sorry, I referred to the wrong type here.  `current` was actually `OutlineViewNode` node so we needed to make a change in `sap.ui.rta.d.ts` file instead.

**Review by voicis — APPROVED** [review]
- Code adheres to development guidelines
- Good test coverage
- Review comments addressed
- Did not test manually, only inspected the code changes

**Review by GDamyanov — APPROVED** [review]
- Code looks good.
- All review comments addressed,
- Excellent test coverage.
- Did not test manually.

**Pull request merged**
